### ISSUE TITLE
Add fields to REST API Media endpoint.

### ISSUE THREAD

**Issue by dnl13** [mundschenk-at/media-credit#34]
Hey, 
thanks for this plugin!
it would be nice if this plugin get self registered to the REST API Media endpoint 
eg: `url.com/wp-json/wp/v2/media/anynumber`
So we can deliver instantly the credits with the media. 

For me i found this solution for theme/function.php
```
add_action('rest_api_init', 'tb_add_custom_rest_fields');
function tb_add_custom_rest_fields() {
    // schema
    $media_credits_schema = array(
        'description'   => 'Copyright of the media item',
        'type'          => 'string',
        'context'       => ['view']
    );

    // registering the field
    register_rest_field(
        'attachment',
        'media_credits',
        [
            'get_callback'      => 'get_media_credits',
            'update_callback'   => null,
            'schema'            => $media_credits_schema
        ]
    );
}

function get_media_credits($object, $field_name, $request) {
    //return 'this-is-a-test';
   	return array(
		'copyright' => get_media_credit(),  
		'copyright_url' => get_media_credit_url(), 
		'copyright_html' => get_media_credit_html(), 
	);
}

```

This will add fields like this
```
...
"media_credits":  {
  "copyright":"© test",
  "copyright_url": "http://www.google.de",
  "copyright_html": "<a href=\"http://www.google.de/\">© test</a>"
},
...
```

Thanks for the plugin and reading!
I hope you are able to add this feature in near future. 
Because of all the headless CMS trends it would be nice. 
regards 
dnl

**Comment by mundschenk-at** [comment]
Yes, you are right, there should be REST API endpoints. I'll tackle this once Gutenberg is in beta (because the shortcode-based integration will need an overhaul anyway).

**Comment by mundschenk-at** [comment]
@dnl13, would you prefer a separate "media_credits" namespace for the fields or just include them in the "meta" namespace (what you get with `register_meta`)?

**Comment by dnl13** [comment]
Hey, 
sorry in the next three weeks i'm not able to do any kind of development. 
I guess it make sense to include it to the "meta", because of other Themes and Addons are able to use this kind as well. 
Don't know exactly (without further deep looking inside) if it can handles 
https://codex.wordpress.org/Function_Reference/wp_read_image_metadata
somehow ?! 
Also i think it should be ok to seperate the namespace for faster access at the api endpoint (maybe with some kind of config ? make true state to get seperate output ?). 
Sorry I#m not a big help atm, but i will put my hands on as soon as possible. 
But thanks for reaching back ! 
regards

**Comment by mundschenk-at** [comment]
I don't think that `wp_read_image_metadata` has got anything to do with the meta in `register_meta`. The former seems to be about reading IPTC and EXIF metadata from image files while the later deals with rows in the `*meta` tables.

**Comment by mundschenk-at** [comment]
After some experimenting, I've chosen to go the `register_rest_field` route to better control the resulting attributes. For now, it uses the `media_credit` attribute with a `rendered` value for the HTML. See https://github.com/mundschenk-at/media-credit/pull/42 for details.

I might move the rest to `raw` to make it clear that those are for editing only. It's not quite the right idiom, though (the API uses `rendered` and `raw` for single attribute values, whereas here we have half a dozen meta values and flags). Also, I might add a `plaintext` attribute to the get the rendered credit without the HTML link.

**Comment by mundschenk-at** [comment]
Implemented by #42.

**Comment by mundschenk-at** [comment]
Version 4.0.0 has been released today. If you are using the REST API in an open source project, I'd be interested in having a look at. I hope the final interface works out OK für your application. 

I retooled the plugin around the JSON representation and I think the code quality is much better now, so it was definitely worthwhile.
